### PR TITLE
Reworked most of the README

### PR DESCRIPTION
**Pull request by fossum**
This rework should look a little better in the README and has some minor fixes in the commands. I can change it from `sudo` based to root login if people are interested.

Might work better to remove Ubuntu references...

### CODE REVIEW

**Comment by JoesCat** [comment]
I somewhat agree with your last sentence and think the entire ubuntu subdirectory could probably be replaced/removed because there are other distros, such as Fedora, and I think they also have a different group mode too.
...probably better said, that maybe the last half-dozen lines (around env==adb) on the mainline script may need a bit more thought to accomidate Ubuntu/Debian (with plugdev), or Fedora/SuSE/etc (using another group)....
...just thinking-out loud....
...if you're willing to wait a bit, I'll try churn-out another pull request for plugdev... (but I'm not running debian/ubuntu, so can't test it)...

File: README.md [hunk 1]
```diff
@@ -1,26 +1,45 @@
 ## Description
-These rules refer to 
+These rules refer to http://developer.android.com/guide/developing/device.html and include many suggestions from the Archlinux and Github Community.
 
-http://developer.android.com/guide/developing/device.html
```
Comment (linux-modder) [inline README.md]: On most linux instances that I've  run into (Fedora guy mostly myself)  there are no default  android udev rules   (would recommend dropping the 'Ubuntu' part tho but the rest of the statement still rings true.

File: README.md [hunk 2]
```diff
@@ -1,26 +1,45 @@
...
+If you're developing on Ubuntu Linux, you need to add a udev rules file that contains a USB configuration for each type of device you want to use for development. In the rules file, each device manufacturer is identified by a unique vendor ID, as specified by the ATTR{idVendor} property. For a list of vendor IDs, see USB Vendor IDs, below.
 
-and include many suggestions from the Archlinux and Github Community
+Use this format to add each vendor to the file(example vendor ID is for HTC):  
+`SUBSYSTEM=="usb", ATTR{idVendor}=="0bb4", MODE="0666", GROUP="plugdev"`
+
+The MODE assignment specifies read/write permissions, and GROUP defines which Unix group owns the device node.
+
```
Comment (linux-modder) [inline README.md]: Also in multi user instances OR under-privileged systems (requiring sudo) group membership to plugdev is  needed in some distros.   The edits here seem to be  aimed at a more  novice 010 user / dev  and  some more advanced conventions seem randomly  disregarded.  Let's line item comment them as they appear or  at the top of this file or  somewhere in the repo have a 'Used / Assumed Conventions.md'   file to alleviate possible new dev /user confusions.

File: README.md [hunk 3]
```diff
@@ -1,26 +1,45 @@
...
+Note: The rule syntax may vary slightly depending on your environment. Consult the udev documentation for your system as needed. For an overview of rule syntax, see this guide to writing udev rules. please refer to 51-android.rules in [ubuntu directory](ubuntu)
```
Comment (linux-modder) [inline README.md]: Again strike out  Ubuntu and replace with Linux / Unix /Mac  or  a simple  'non -windows'  env

File: README.md [hunk 4]
```diff
@@ -1,26 +1,45 @@
...
 
 ## Installation
-If you're developing on Ubuntu Linux, you need to add a udev rules file that contains a USB configuration for each type of device you want to use for development. In the rules file, each device manufacturer is identified by a unique vendor ID, as specified by the ATTR{idVendor} property. For a list of vendor IDs, see USB Vendor IDs, below. To set up device detection on Ubuntu Linux:
-Log in as root and create this file: /etc/udev/rules.d/51-android.rules.
-Use this format to add each vendor to the file:
-SUBSYSTEM=="usb", ATTR{idVendor}=="0bb4", MODE="0666", GROUP="plugdev"
```
Comment (linux-modder) [inline README.md]: also for  the less advanced could  have this inserted.   So there is less likelihood of file corruption and  uses a much more common Copy operation and allows for a master and modified version for the developer with less reliance on more advanced knowledge.

 
git clone <repo-url>
cd  android-udev-rules
sudo cp 51-android-udev-rules /etc/udev/rules.d/  
cd

File: README.md [hunk 5]
```diff
@@ -1,26 +1,45 @@
...
-Try group plugdev rather than plugindev. Then restart udev:
+```sh
+# Clone the repo (replace repo-url with the Github URL)
+git clone <repo-url>
+# Create a sym-link to the rules file
+sudo ln -s ./android-udev-rules/51-android.rules /etc/udev/rules.d/
+# Change file permissions
+sudo chmod a+r /etc/udev/rules.d/51-android.rules
+# Restart UDEV
 sudo udevadm control --reload-rules
```
Comment (linux-modder) [inline README.md]: On 14.x + and most other linux rigs these days a simple  unplug  <device> and  re-insert forces the reload of rules and more  new user /dev friendly (imo)

File: README.md [hunk 6]
```diff
@@ -1,26 +1,45 @@
...
 sudo service udev restart
```
Comment (linux-modder) [inline README.md]: Systemd users:

sudo systemctl restart udev
sudo systemctl enable (or  simply  start) adb

**Comment by JoesCat** [comment]
This is part of the "thinking out loud", but this is the idea here...
```c
 
 # Enable device as a user device if found (add an "android" SYMLINK)
 ENV{adb_user}=="yes", MODE="0660", GROUP="adbusers", TAG+="uaccess", SYMLINK+="android"
+# NOTE: Debian/Ubuntu based distros use "plugdev"
+# This next line fails for distros that don't have "plugdev", but that is okay
+ENV{adb_user}=="yes", GROUP="plugdev"
 
 # Devices listed here {begin...end} are connected by USB
 LABEL="android_usb_rules_end"
```
...what should happen above is that if the distro doesn't have a plugdev group, then it shouldn't change. Another possibility is to try this...
```c
 
 # Enable device as a user device if found (add an "android" SYMLINK)
 ENV{adb_user}=="yes", MODE="0660", GROUP="adbusers", TAG+="uaccess", SYMLINK+="android"
+# NOTE: Debian/Ubuntu based distros use "plugdev"
+# This next line fails for distros that don't have "plugdev", but that is okay
+ENV{adb_user}=="yes", ENV{adb_adb}!="yes", GROUP="plugdev"
 
 # Devices listed here {begin...end} are connected by USB
 LABEL="android_usb_rules_end"
```
...however, with this idea above, some devices listed above don't have a separate debug/developer mode, and will need to be modified accordingly.
If following the Ubuntu file, maybe MODE="0660" becomes MODE="0666", but that basically defeats the security model of using plugdev in the first place.

I leave it to you to include this in your patch (so it stays together with the patch).

**Pull request merged**